### PR TITLE
Added ban method to guild member object

### PR DESCRIPTION
**Pull request by ahnaf-zamil**
### Summary
I have added a ban method to the guild member object that allows you to ban a member without using the bot's REST client
and by simply using `await member.ban()`. It takes the same optional parameters as `hikari.api.rest.RESTClient.ban_user`.

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.
- [x] I used a test bot and the bot could ban a member using the code I have added.

### Related issues

#37

### CODE REVIEW

File: hikari/api/rest.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 __all__: typing.List[str] = ["ConnectorFactory", "RESTClient"]
 
 import abc
+import builtins
```
Comment (FasterSpeeding) [inline hikari/api/rest.py:28]: don't need to import builtins here, just use `int` in the type hint (docstring type still has to be builtins.int though)

File: hikari/guilds.py [hunk 2]
```diff
@@ -439,6 +440,51 @@ async def fetch_self(self) -> Member:
         """
         return await self.app.rest.fetch_member(self.guild_id, self.id)
 
+    async def ban(
+        self,
+        *,
+        delete_message_days: undefined.UndefinedOr[builtins.int] = undefined.UNDEFINED,
```
Comment (FasterSpeeding) [inline hikari/guilds.py:446]: don't need to import builtins here, just use `int` in the type hint (docstring type still has to be builtins.int though)

File: hikari/impl/rest.py [hunk 3]
```diff
@@ -34,6 +34,7 @@
 ]
 
 import asyncio
+import builtins
```
Comment (FasterSpeeding) [inline hikari/impl/rest.py:37]: don't need to import builtins here, just use `int` in the type hint

File: tests/hikari/test_guilds.py [hunk 4]
```diff
@@ -239,6 +239,11 @@ def test_format_avatar(self, model, mock_user):
         mock_user.format_avatar.assert_called_once_with(ext="png", size=4096)
         assert result is mock_user.format_avatar.return_value
 
+    async def test_ban(self, model, mock_user):
+        result = await mock_user.ban(reason="cuz u was meen")
```
Comment (FasterSpeeding) [inline tests/hikari/test_guilds.py:243]: to be more thorougher we'd probably want to test it with all the optional arguments being passed in 1 test and none of them being passed in another but this isn't too important

**Review by FasterSpeeding — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 5]
```diff
@@ -63,3 +63,7 @@ __init__.pyi
 
 # Logs
 *.log
+
+# Lightbulb and test bot
+lightbulb/
```
Comment (FasterSpeeding) [inline .gitignore:68]: why is light bulb in your testing environment? couldn't you just use `pip install -e path_to_local_project`?

File: hikari/api/rest.py (hunk 1 above)
Comment (ahnaf-zamil) [reply hikari/api/rest.py:28]: @davfsa told me to use `builtins.int` there

File: hikari/guilds.py (hunk 2 above)
Comment (ahnaf-zamil) [reply hikari/guilds.py:446]: @FasterSpeeding  i don't get it. Davfsa told me to put builtins.int there. Then if I put int in the type but builtins.int in docstring, that doesn't make any sense

File: tests/hikari/test_guilds.py (hunk 4 above)
Comment (ahnaf-zamil) [reply tests/hikari/test_guilds.py:243]: @FasterSpeeding  I tested it with a test bot and it's working fine with or without reason

File: .gitignore (hunk 5 above)
Comment (ahnaf-zamil) [reply .gitignore:68]: cuz i was having issues with adding it to the test env

File: hikari/api/rest.py (hunk 1 above)
Comment (davfsa) [reply hikari/api/rest.py]: I told you to use it on the docstring, not to actually import it. The pdoc template we have uses `builtins.` to indicate that it has to look in the builtins to link it

Comment (ahnaf-zamil) [reply hikari/api/rest.py]: i did put it there

File: .gitignore [hunk 6]
```diff
@@ -63,3 +63,7 @@ __init__.pyi
...
+test-bot.py
```
Comment (davfsa) [inline .gitignore:69]: Could you please remove these? They have nothing to do with the project

**Review by davfsa — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 6 above)
Comment (ahnaf-zamil) [reply .gitignore:69]: i put them here just for testing, i will remove it if you want to

Comment (davfsa) [reply .gitignore:69]: Yes please

File: .gitignore [hunk 7]
```diff
@@ -63,3 +63,4 @@ __init__.pyi
 
 # Logs
 *.log
+
```
Comment (davfsa) [inline .gitignore:66]: Extra line here

File: tests/hikari/test_guilds.py [hunk 8]
```diff
@@ -239,6 +239,11 @@ def test_format_avatar(self, model, mock_user):
...
+        mock_user.ban.assert_awaited_once_with(reason="cuz u was meen")
+        assert result is mock_user.ban.return_value
+
```
Comment (davfsa) [inline tests/hikari/test_guilds.py:246]: This test is wrong, you are mocking `ban` and then just checking that you are calling it.

You should not mock ban and check if it calls `app.rest.ban_user` with the correct args

File: hikari/guilds.py [hunk 9]
```diff
@@ -439,6 +439,51 @@ async def fetch_self(self) -> Member:
...
+            rate-limits automatically. This includes most bucket-specific
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+        return await self.app.rest.ban_member(
```
Comment (davfsa) [inline hikari/guilds.py:483]:
```suggestion
        return await self.app.rest.ban_user(
```

**Review by davfsa — CHANGES_REQUESTED** [review]

File: tests/hikari/test_guilds.py (hunk 8 above)
Comment (ahnaf-zamil) [reply tests/hikari/test_guilds.py:246]: I don't understand what u mean. What do I have to put on the test? I'm pretty new to pytest and testing in general. So if you could tell me literally what to do, I could get some ideas

File: hikari/guilds.py [hunk 10]
```diff
@@ -439,6 +439,51 @@ async def fetch_self(self) -> Member:
...
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+        return await self.app.rest.ban_user(
+            self.guild_id, self.id, delete_message_days=delete_message_days, reason=reason
+        )
```
Comment (davfsa) [inline hikari/guilds.py:485]: This return is redundant

```suggestion
        await self.app.rest.ban_user(
            self.guild_id, self.id, delete_message_days=delete_message_days, reason=reason
        )
```

**Review by davfsa — CHANGES_REQUESTED** [review]

**Review by davfsa — APPROVED** [review]

**Review by davfsa — APPROVED** [review]

**Comment by davfsa** [comment]
Force merging due to codeclimate not working
